lib/strings.nix

```
@@ -157,6 +157,76 @@ rec {
   */
   replicate = n: s: concatStrings (lib.lists.replicate n s);
 
+  /*
+    Remove leading and trailing whitespace from a string.
+
+    Whitespace is defined as any of the following characters:
+      " ", "\t" "\r" "\n"
+
+    Type: trim :: string -> string
+
+    Example:
+      trim "   hello, world!   "
+      => "hello, world!"
+  */
+  trim = trim' {
+    start = true;
+    end = true;
+  };
+
+  /*
+    Remove leading and/or trailing whitespace from a string.
+
+    Whitespace is defined as any of the following characters:
+      " ", "\t" "\r" "\n"
+
+    Type: trim' :: Attrs -> string -> string
+
+    Example:
+      trim' { start = true; } "   hello, world!   "}
+      => "hello, world!   "
+      trim' { end = true; } "   hello, world!   "}
+      => "   hello, world!"
+  */
+  trim' =
```

```suggestion
  trimWith =
```

This is emerging as a convention :)

---

Sure thing, done.